saleor/checkout/complete_checkout.py

```diff
@@ -428,4 +432,9 @@ def complete_checkout(
             gateway.payment_refund_or_void(payment)
             raise ValidationError(f"Insufficient product stock: {e.item}", code=e.code)
 
+        # if there is no payment and no errors occurred it means that total value is 0
+        # and is paid from definition
+        if not payment:
```

Comment:
Wouldn't something like `order.total==0` be more meaningful?

Reply:
Good point. It might be better.